src/TSID/include/BipedalLocomotion/TSID/QPFixedBaseTSID.h

```diff
@@ -0,0 +1,143 @@
+ */
+class QPFixedBaseTSID : public TaskSpaceInverseDynamics
+{
+   /**
+     * Private implementation
+     */
+    struct Impl;
+    std::unique_ptr<Impl> m_pimpl;
+
+    bool createBaseSE3Task();
+    bool createDynamicsTask();
+
+public:
+
+    /**
+     * Constructor.
+     */
+    QPFixedBaseTSID();
+
+    /**
+     * Destructor.
+     */
+    ~QPFixedBaseTSID();
+
+    /**
+     * Add a linear task in the fixed base TSID
+     * @param task pointer to a given linear task
+     * @param priority Priority associated to the task. The lower the number the higher the
+     * priority.
+     * @param weight weight associated to the task. This parameter is optional. The default value is
+     * an object that does not contain any value. So is an invalid weight.
+     * @note currently we support only task with priority 0 or 1. If the priority is set to 0 the
+     * task will be considered as a constraint. In this case the weight is not required.
+     * @warning The QPFixedBaseTSID cannot handle inequality tasks (please check
+     * Task::Type) with priority equal to 1.
+     * @return true if the task has been added to the TSID.
+     */
+    bool addTask(std::shared_ptr<System::LinearTask> task,
+                 const std::string& taskName,
+                 std::size_t priority,
+                 std::optional<Eigen::Ref<const Eigen::VectorXd>> weight = {}) override;
+
+    /**
+     * Get a vector containing the name of the tasks.
+     * @return an std::vector containing all the names associated to the tasks
+     */
+    std::vector<std::string> getTaskNames() const override;
+
+    /**
+     * Get a specific task
+     * @param name name associated to the task.
+     * @return a weak ptr associated to an existing task in the TSID. If the task does not exist a
+     * nullptr is returned.
+     */
+    std::weak_ptr<System::LinearTask> getTask(const std::string& name) const override;
+
+    /**
+     * Initialize the TSID algorithm.
+     * @param handler pointer to the IParametersHandler interface.h
+     * @note the following parameters are required by the class
+     * |            Parameter Name            |   Type   |                                             Description                                            | Mandatory |
+     * |:------------------------------------:|:--------:|:--------------------------------------------------------------------------------------------------:|:---------:|
+     * |  `robot_acceleration_variable_name`  | `string` | Name of the variable contained in `VariablesHandler` describing the generalized robot acceleration |    Yes    |
+     * |    `robot_torque_variable_name`      | `string` |         Name of the variable contained in `VariablesHandler` describing the robot torque           |    Yes    |
+     * | `robot_contact_wrench_variable_name` | `string` |      Name of the variable contained in `VariablesHandler` describing the robot contact wrench      |    Yes    |
```

Comment:
I would just pass the name of the link that should be considered in contact.
Notice that in your implementation you are assuming that the root link is in contact. If you do not want to add the possibility to have a fixed frame different from you should remove `robot_contact_wrench_variable_name`